src/common/modals/InstanceManager/Overview.js

```diff
@@ -501,9 +530,56 @@ const Overview = ({ instanceName, background, manifest }) => {
                   debouncedArgumentsUpdate.callback(e.target.value);
                 }}
               />
-              <JavaArgumentsResetButton onClick={resetJavaArguments}>
+              <JavaResetButton onClick={resetJavaArguments}>
+                <FontAwesomeIcon icon={faUndo} />
+              </JavaResetButton>
+            </JavaManagerRow>
+          )}
+          <JavaManagerRow>
+            <div>Custom Java Path</div>
+            <Switch
+              checked={customJavaPathSwitch}
+              onChange={v => {
+                setCustomJavaPathSwitch(v);
+
+                if (!v) {
+                  dispatch(
+                    updateInstanceConfig(instanceName, prev =>
+                      omit(prev, ['customJavaPath'])
+                    )
+                  );
+                } else if (v) {
+                  resetCustomJavaPath();
+                }
+              }}
+            />
+          </JavaManagerRow>
+          {customJavaPathSwitch && (
+            <JavaManagerRow>
+              <Input
+                value={customJavaPath}
+                onChange={e => {
+                  setCustomJavaPath(e.target.value);
```

Comment:
What if you type the path and don't actually use the button? The changes won't be saved to redux...